src/stdlib/encode_punycode.rs

```diff
@@ -83,6 +83,10 @@ impl FunctionExpression for EncodePunycodeFn {
                 .map_err(|errors| format!("unable to encode to punycode: {errors}"))?;
             Ok(encoded.into())
         } else {
+            if string.is_ascii() {
```

Comment:
Same question ([as this](https://github.com/vectordotdev/vrl/pull/1104/files#r1831688371)) here.

Reply:
I think we can safely do it, since it is already done in the other branch. `idna` crate does a similar check right at the start (it also looks for the punycode prefix `xn--`, but that is only needed for validation and it is disabled in this branch), but I have noticed a slight difference - the library also normalizes uppercase characters to lowercase, so it only checks for lowercase characters.

I think I also need to fix this behavior now in general - to also replace uppercase character to lowercase when `validate: false`, for consistency with `validate: true` branch.

Should that fix be done here, or a separate PR?

Reply:
> Should that fix be done here, or a separate PR?
I would prefer a separate PR but both are acceptable.

Reply:
I think separate PR makes more sense too, so I will do it like that.